File: generated-sources/javascript-closure-angular/mojang-api/API/Client/ChangeSkinRequest.js
goog.provide('API.Client.ChangeSkinRequest');

/**
 * Request Mojang to download a skin from an URL and apply to the player
 * @record
 */
API.Client.ChangeSkinRequest = function() {}

/**
 * @type {!API.Client.SkinModel}
 * @export
 */
API.Client.ChangeSkinRequest.prototype.model;

/**
 * The URL which Mojang servers will download and apply the skin
 * @type {!string}
 * @export
 */
API.Client.ChangeSkinRequest.prototype.url;

File: generated-sources/kotlin-spring/mojang-api/src/main/kotlin/com/github/asyncmc/mojang/api/kotlin/spring/model/ChangeSkinRequest.kt
package com.github.asyncmc.mojang.api.kotlin.spring.model

import java.util.Objects
import com.fasterxml.jackson.annotation.JsonProperty
import com.github.asyncmc.mojang.api.kotlin.spring.model.SkinModel
import javax.validation.Valid
import javax.validation.constraints.*

/**
 * Request Mojang to download a skin from an URL and apply to the player
 * @param model 
 * @param url The URL which Mojang servers will download and apply the skin
 */
data class ChangeSkinRequest (

        @get:NotNull 
        @JsonProperty("url") val url: String,

        @JsonProperty("model") val model: SkinModel? = null
) {

}

